gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ReferralRequestMapper.java

```diff
@@ -0,0 +1,230 @@
+            .map(RCMRMT030101UK04AgentRef::getId)
+            .map(II::getRoot)
+            .findFirst();
+    }
+
+    private boolean hasAgentReference(RCMRMT030101UK04Participant participant) {
+        return participant.getAgentRef() != null && participant.getAgentRef().getId() != null;
+    }
+
+    private boolean hasTypeCode(RCMRMT030101UK04Participant participant, String typeCode) {
+        return !participant.getTypeCode().isEmpty() && participant.getTypeCode().get(0).equals(typeCode);
+    }
+
+    private boolean isNotNullFlavourParticipant(RCMRMT030101UK04Participant participant) {
+        return participant.getNullFlavor() == null;
+    }
+
+    private List<Annotation> getNotes(RCMRMT030101UK04RequestStatement requestStatement) {
+        var priority = getPriorityText(requestStatement.getPriorityCode());
+        var actionDate = getActionDateText(requestStatement.getEffectiveTime());
+        var text = requestStatement.getText();
+
+        var notes = new ArrayList<Annotation>();
+
+        if (StringUtils.isNotEmpty(priority)) {
+            notes.add(new Annotation().setText(priority));
+        }
+
+        if (StringUtils.isNotEmpty(actionDate)) {
+            notes.add(new Annotation().setText(actionDate));
+        }
+
+        if (StringUtils.isNotEmpty(text)) {
+            notes.add(new Annotation().setText(text));
+        }
+
+        return notes;
+    }
+
+    private String getPriorityText(CV priorityCode) {
+        if (priorityCode != null) {
+            if (StringUtils.isNotEmpty(priorityCode.getOriginalText())) {
+                return PRIORITY_PREFIX + priorityCode.getOriginalText();
+            } else if (StringUtils.isNotEmpty(priorityCode.getDisplayName())) {
+                return PRIORITY_PREFIX + priorityCode.getDisplayName();
+            }
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String getActionDateText(IVLTS effectiveTime) {
+        if (hasEffectiveTimeValue(effectiveTime)) {
+            return ACTION_DATE_PREFIX + DateFormatUtil.parse(effectiveTime.getCenter().getValue()).asStringValue();
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    private boolean hasEffectiveTimeValue(IVLTS effectiveTime) {
+        return effectiveTime != null && effectiveTime.getCenter() != null && effectiveTime.getCenter().getValue() != null;
+    }
+
+    private ReferralRequest createRequestStatement(String id, Identifier identifier, List<Annotation> notes, CodeableConcept reasonCode,
+        Date authoredOn, Reference recipient, Reference requester) {
+        var referralRequest = new ReferralRequest();
+
+        referralRequest.setId(id);
+        referralRequest.getMeta().getProfile().add(new UriType(META_PROFILE));
+        referralRequest.getIdentifier().add(identifier);
+        referralRequest.setStatus(ReferralRequestStatus.UNKNOWN);
+        referralRequest.setIntent(ReferralCategory.ORDER);
+
+        if (requester != null) {
```

Comment:
Could you check please (for example using unit tests), if setting null values for all of those fields can actually fail? If not, then maybe there's no need to check if they are null before setting them. Thanks!

Reply:
You're right, these null checks aren't needed! 😄